src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java

```diff
@@ -0,0 +1,13 @@
+package org.dasein.cloud.network;
+
+import org.dasein.cloud.AbstractProviderService;
+import org.dasein.cloud.CloudProvider;
+
+
+public abstract class AbstractVPNSupport<T extends CloudProvider> extends AbstractProviderService<T> implements VPNSupport {
+
+    protected AbstractVPNSupport(T provider) {
+        super(provider);
+    }
+
```

Comment:
There's little point in this abstract class unless it implements most of the methods from VPNSupport with `throws OperationNotSupportedException("Blah is not currently implemented in " + getProvider().getCloudName());`. The only two methods it shouldn't implement is `isSubscribed()` and `getCapabilities()`;